packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -221,15 +229,41 @@ const ProductDetail = () => {
         }
 
         return true
-    }, [product, productSetSelection])
+    }, [product, childProductSelection])
 
     const handleProductSetAddToCart = () => {
         // Get all the selected products, and pass them to the addToCart handler which
         // accepts an array.
-        const productSelectionValues = Object.values(productSetSelection)
+        const productSelectionValues = Object.values(childProductSelection)
         return handleAddToCart(productSelectionValues)
     }
 
+    /**************** Product Bundle Handlers ****************/
+
+    // Top level bundle does not have variants
+    const handleProductBundleAddToCart = async (variant, selectedQuantity) => {
+        try {
+            const productItems = [
+                {
+                    productId: product.id,
+                    price: product.price,
+                    quantity: selectedQuantity
+                }
+            ]
```

Comment:
We'll need to pass in the child variant ids, whenever a bundle is added to cart. Something like this:

```suggestion
            const bundledProductItems = Object.values(childProductSelection).map((child) => {
                const productId = child.variant.productId
                const quantity = child.quantity
                return {
                    productId,
                    quantity
                }
            })

            const productItems = [
                {
                    productId: product.id,
                    price: product.price,
                    quantity: selectedQuantity,
                    bundledProductItems
                }
            ]
```